agent/agent.go

```diff
@@ -20,24 +20,30 @@ const (
 
 // Agent runs recipes for specified plugins.
 type Agent struct {
-	extractorFactory *registry.ExtractorFactory
-	processorFactory *registry.ProcessorFactory
-	sinkFactory      *registry.SinkFactory
-	monitor          Monitor
-	logger           log.Logger
+	extractorFactory     *registry.ExtractorFactory
+	processorFactory     *registry.ProcessorFactory
+	sinkFactory          *registry.SinkFactory
+	monitor              Monitor
+	logger               log.Logger
+	retryTimes           int
```

Comment:
@StewartJingga If this represents no of retries. Then we can call it `maxRetries` WDYT?